apps/dashboard/src/hooks/use-has-permission.tsx

```diff
@@ -0,0 +1,56 @@
+import { useMemo } from 'react';
+import { useFeatureFlag } from '@/hooks/use-feature-flag';
+import {
+  FeatureFlagsKeysEnum,
+  ApiServiceLevelEnum,
+  FeatureNameEnum,
+  getFeatureForTierAsBoolean,
+  GetSubscriptionDto,
+} from '@novu/shared';
+import { useFetchSubscription } from '@/hooks/use-fetch-subscription';
+import { CheckAuthorizationWithCustomPermissions } from '@clerk/types';
+import { useAuth } from '@clerk/clerk-react';
+
+function isRbacEnabled(isRbacFlagEnabled: boolean, subscription: GetSubscriptionDto | undefined): boolean {
+  const isEligibleTier =
+    subscription &&
+    !subscription.trial.isActive &&
+    subscription.apiServiceLevel !== ApiServiceLevelEnum.FREE &&
+    subscription.apiServiceLevel !== ApiServiceLevelEnum.PRO;
```

Comment:
I remember we had some shared function to handle feature per tier, maybe we can reuse it here?